tests/UnifiedSpecTests/Operation.php

```diff
@@ -504,74 +651,108 @@ private function executeForTestRunner()
     {
         $args = $this->prepareArguments();
 
-        if (array_key_exists('client', $args)) {
-            assertIsString($args['client']);
-            $args['client'] = $this->entityMap->getClient($args['client']);
-        }
-
         switch ($this->name) {
             case 'assertCollectionExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 $database = $this->context->getInternalClient()->selectDatabase($args['databaseName']);
                 assertContains($args['collectionName'], $database->listCollectionNames());
                 break;
             case 'assertCollectionNotExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 $database = $this->context->getInternalClient()->selectDatabase($args['databaseName']);
                 assertNotContains($args['collectionName'], $database->listCollectionNames());
                 break;
             case 'assertIndexExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
+                assertArrayHasKey('indexName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 assertIsString($args['indexName']);
                 assertContains($args['indexName'], $this->getIndexNames($args['databaseName'], $args['collectionName']));
                 break;
             case 'assertIndexNotExists':
+                assertArrayHasKey('databaseName', $args);
+                assertArrayHasKey('collectionName', $args);
+                assertArrayHasKey('indexName', $args);
                 assertIsString($args['databaseName']);
                 assertIsString($args['collectionName']);
                 assertIsString($args['indexName']);
                 assertNotContains($args['indexName'], $this->getIndexNames($args['databaseName'], $args['collectionName']));
                 break;
             case 'assertSameLsidOnLastTwoCommands':
+                /* Context::getEventObserverForClient() requires the client ID.
+                 * Avoid checking $args['client'], which is already resolved. */
+                assertArrayHasKey('client', $this->arguments);
```

Comment:
This is one of the few places we assert on `$this->arguments` instead of `$args`, because we need the unresolved entity ID for the Context method. We also do this below for `assertSessionDirty`.